### PR TITLE
chore: skip doc build when property specified

### PR DESCRIPTION
**Pull request by miurahr**
- Introduce "forceSkipDocumentBuild" property which indicate to skip document build steps.
- Developer can specify Gradle property with "-P" option, such as "gradle -PforceSkipDocumentBuild=1" or via environment variable ORG_GRADLE_PROJECT_forceSkipDocumentBuild=1 
- see https://docs.gradle.org/current/userguide/project_properties.html

## Pull request type

- Build and release changes -> [build/release]

## Which ticket is resolved?

Dev-ML
https://sourceforge.net/p/omegat/mailman/omegat-development/thread/F86A5D44-20F7-492D-9F12-98E1C7A9DA97%40traduction-libre.org/#msg58712422

## What does this PR change?

- add document build condition to check property in addition to check of existence of docker command

### CODE REVIEW

**Comment by brandelune** [comment]
```
Could not compile build file '/Users/suzume/Documents/Repositories/omegat.brandelune/build.gradle'.
> startup failed:
  build file '/Users/suzume/Documents/Repositories/omegat.brandelune/build.gradle': 618: Unexpected input: '{' @ line 618, column 15.
     distributions {
```

**Comment by miurahr** [comment]
When execute `./gradlew installDist -PforceSkipDocumentBuild` then you can observe

```

> Configure project :
Detected revision 41865e257

> Task :firstStepsCa SKIPPED
Specified forceSkipDocumentBuild property

> Task :firstStepsDe SKIPPED
Specified forceSkipDocumentBuild property

> Task :firstStepsEn SKIPPED
Specified forceSkipDocumentBuild property

> Task :firstStepsFr SKIPPED
Specified forceSkipDocumentBuild property

> Task :firstStepsJa SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideBe SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideCs SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideCy SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideEl SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideEs SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideEu SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideFi SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideGl SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideHr SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideHu SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideIa SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideIt SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideKo SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideMfe SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideNl SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideNo SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuidePt SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuidePt_BR SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideRu SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideSc SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideSv SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideTr SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideUk SKIPPED
Specified forceSkipDocumentBuild property

> Task :instantStartGuideZh_CN SKIPPED
Specified forceSkipDocumentBuild property

> Task :firstSteps UP-TO-DATE
> Task :compileJava UP-TO-DATE
> Task :processResources UP-TO-DATE
> Task :classes UP-TO-DATE
> Task :jar UP-TO-DATE

> Task :manualHtmlEn SKIPPED
Specified forceSkipDocumentBuild property

> Task :manualZipEn UP-TO-DATE

> Task :manualHtmlFr SKIPPED
Specified forceSkipDocumentBuild property

> Task :manualZipFr UP-TO-DATE

> Task :manualHtmlJa SKIPPED
Specified forceSkipDocumentBuild property

> Task :manualZipJa UP-TO-DATE
> Task :manualZips UP-TO-DATE
> Task :startScripts SKIPPED
> Task :aligner:compileJava UP-TO-DATE
> Task :aligner:processResources UP-TO-DATE
> Task :aligner:classes UP-TO-DATE
> Task :aligner:jar UP-TO-DATE
> Task :machinetranslators:compileJava NO-SOURCE
> Task :machinetranslators:processResources NO-SOURCE
> Task :machinetranslators:classes UP-TO-DATE
> Task :machinetranslators:jar SKIPPED
> Task :spellchecker:compileJava NO-SOURCE
> Task :spellchecker:processResources NO-SOURCE
> Task :spellchecker:classes UP-TO-DATE
> Task :spellchecker:jar SKIPPED
> Task :theme:compileJava UP-TO-DATE
> Task :theme:processResources NO-SOURCE
> Task :theme:classes UP-TO-DATE
> Task :theme:jar UP-TO-DATE
> Task :tipoftheday:compileJava UP-TO-DATE
> Task :tipoftheday:processResources UP-TO-DATE
> Task :tipoftheday:classes UP-TO-DATE
> Task :tipoftheday:jar UP-TO-DATE
> Task :machinetranslators:apertium:compileJava UP-TO-DATE
> Task :machinetranslators:apertium:processResources UP-TO-DATE
> Task :machinetranslators:apertium:classes UP-TO-DATE
> Task :machinetranslators:apertium:jar UP-TO-DATE
> Task :machinetranslators:belazar:compileJava UP-TO-DATE
> Task :machinetranslators:belazar:processResources UP-TO-DATE
> Task :machinetranslators:belazar:classes UP-TO-DATE
> Task :machinetranslators:belazar:jar UP-TO-DATE
> Task :machinetranslators:deepl:compileJava UP-TO-DATE
> Task :machinetranslators:deepl:processResources UP-TO-DATE
> Task :machinetranslators:deepl:classes UP-TO-DATE
> Task :machinetranslators:deepl:jar UP-TO-DATE
> Task :machinetranslators:google:compileJava UP-TO-DATE
> Task :machinetranslators:google:processResources UP-TO-DATE
> Task :machinetranslators:google:classes UP-TO-DATE
> Task :machinetranslators:google:jar UP-TO-DATE
> Task :machinetranslators:ibmwatson:compileJava UP-TO-DATE
> Task :machinetranslators:ibmwatson:processResources UP-TO-DATE
> Task :machinetranslators:ibmwatson:classes UP-TO-DATE
> Task :machinetranslators:ibmwatson:jar UP-TO-DATE
> Task :machinetranslators:mymemory:compileJava UP-TO-DATE
> Task :machinetranslators:mymemory:processResources UP-TO-DATE
> Task :machinetranslators:mymemory:classes UP-TO-DATE
> Task :machinetranslators:mymemory:jar UP-TO-DATE
> Task :machinetranslators:yandex:compileJava UP-TO-DATE
> Task :machinetranslators:yandex:processResources UP-TO-DATE
> Task :machinetranslators:yandex:classes UP-TO-DATE
> Task :machinetranslators:yandex:jar UP-TO-DATE
> Task :spellchecker:hunspell-jmyspell:compileJava UP-TO-DATE
> Task :spellchecker:hunspell-jmyspell:processResources UP-TO-DATE
> Task :spellchecker:hunspell-jmyspell:classes UP-TO-DATE
> Task :spellchecker:hunspell-jmyspell:jar UP-TO-DATE
> Task :installDist
```

**Comment by brandelune** [comment]
Ok. It looks like it works. Thank you.

File: docs_devel/docs/02.HowToBuild.md [hunk 1]
```diff
@@ -17,13 +16,27 @@ produce Javadoc by running `gradlew javadoc`, or browse online:
...
 
 ## Configuring Build Tasks
 
 Some build tasks, such as signed installers, require additional configuration
 via a `local.properties` file placed at the root of the source tree. See
 `local.properties.example`.
 
+When specified a Gradle property `forceSkipDocumentBuild`, the build task 
+will skip build processes to generate greeting documents, `FirstStep.html`
+in languages, and bundled manuals. You can edit `local.properties` or
```
Comment (SlyJackHammer) [inline docs_devel/docs/02.HowToBuild.md:39]: Was there supposed to be something else here? It seems incomplete:
"You can edit `local.properties` or"

I'm starting out on OmegaT development, trying to understand the codebase and build it myself.

Comment (miurahr) [reply docs_devel/docs/02.HowToBuild.md:39]: Thanks for pointing out missing sentence.
"You can edit local.properties or $HOME/.gradle./gradle.properties."

Comment (miurahr) [reply docs_devel/docs/02.HowToBuild.md:39]: > I'm starting out on OmegaT development, trying to understand the codebase and build it myself.

You can also check it on readthedocs website.
https://omegat.readthedocs.io/en/latest/02.HowToBuild/

Comment (SlyJackHammer) [reply docs_devel/docs/02.HowToBuild.md:39]: Thank you!

**Pull request merged**